File: solutions/c/collatz-conjecture/1/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int start) {
    if (start <= 0) {return ERROR_VALUE;}
    int end = 1;
    int cur = start;
    int index = 0;
    while (cur != end)
    {
        if (cur % 2 == 0) {
            cur /= 2;
            index++;
        } else {
            cur = (cur * 3) + 1;
            index++;
        }

    }
    return index;
}

File: solutions/go/collatz-conjecture/1/collatz_conjecture.go
package collatzconjecture

import "errors"

func CollatzConjecture(n int) (steps int, err error) {

	if n < 1 {
		return -1, errors.New("zero and negative numbers are unsupported")
	}
	for n != 1 {
		steps++
		if n&1 == 0 {
			n /= 2
		} else {
			n = 3*n + 1
		}
	}
	return
}
